compiler/src/language_server/sourcetree.re

```diff
@@ -427,11 +441,7 @@ module Sourcetree: Sourcetree = {
             switch (get_type_path(pat.pat_type)) {
             | Some(path) =>
               let decl = Env.find_type(path, pat.pat_env);
-              if (decl.type_loc == Location.dummy_loc) {
-                None;
-              } else {
-                Some(decl.type_loc);
-              };
+              Some(decl.type_loc);
```

Comment:
We need feedback from @ospencer because I'm pretty sure this had some purpose that you are missing.

Reply:
Yeah, I'd likely put this back. It's an option just to be specific about if there's a location or not. Changing this back shouldn't affect anything.

Reply:
I reapplied that code